src/storage/types/src/parse.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! Common, performance-oriented mechanisms of parsing byte strings into various types
```

Comment:
Yeah, I think this should probably be in protocol-common?

Reply:
moved it